### PR TITLE
fix #2092.

### LINKED ISSUE #2092: appid大小不一致导致不能拉取配置更新
当appid中存在字母时，如果客户端配置和portal中配置的appid大小写不一致，仍然可以获取配置，当配置发生更改，客户端无法收到最新的配置推送。建议appid大小写不敏感或禁用字母，否则容易产生莫名其妙的问题。

### PR DESCRIPTION
**Pull request by Accelerater**
通过比较hashCode的方式。就不会出现
a.compareTo(b) = 1, b.compareTo(a) = 1，a > b 且 b > a这样的结果了.
@nobodyiam 你觉得这样可行么？

### CODE REVIEW

File: apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/wrapper/DeferredResultWrapper.java [hunk 1]
```diff
@@ -64,4 +64,10 @@ public void setResult(List<ApolloConfigNotification> notifications) {
   public DeferredResult<ResponseEntity<List<ApolloConfigNotification>>> getResult() {
     return result;
   }
+
+  @Override
+  public int compareTo(DeferredResultWrapper deferredResultWrapper) {
+    long result = this.hashCode() - deferredResultWrapper.hashCode();
+    return Long.compare(result, 0L);
```
Comment (JaredTan95) [inline apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/wrapper/DeferredResultWrapper.java]: `Integer.compare(,)` is enough?

File: apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/wrapper/DeferredResultWrapper.java [hunk 2]
```diff
@@ -64,4 +64,10 @@ public void setResult(List<ApolloConfigNotification> notifications) {
   public DeferredResult<ResponseEntity<List<ApolloConfigNotification>>> getResult() {
     return result;
   }
+
+  @Override
+  public int compareTo(DeferredResultWrapper deferredResultWrapper) {
+    long result = this.hashCode() - deferredResultWrapper.hashCode();
```
Comment (nobodyiam) [inline apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/wrapper/DeferredResultWrapper.java]: check whether deferredResultWrapper is null?

if not null, then return the comparison result between the two hashcode should be ok.

**Comment by Accelerater** [comment]
加上了@NonNull 注解

File: apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/wrapper/DeferredResultWrapper.java (hunk 2 above)
Comment (Accelerater) [reply apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/wrapper/DeferredResultWrapper.java]: 添加了 @ NonNull 注解。
下面是compareTo接口的信息:
image

File: apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/wrapper/DeferredResultWrapper.java [hunk 3]
```diff
@@ -64,4 +65,10 @@ public void setResult(List<ApolloConfigNotification> notifications) {
   public DeferredResult<ResponseEntity<List<ApolloConfigNotification>>> getResult() {
     return result;
   }
+
+  @Override
+  public int compareTo(@NonNull DeferredResultWrapper deferredResultWrapper) {
+    int result = this.hashCode() - deferredResultWrapper.hashCode();
```
Comment (nobodyiam) [inline apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/wrapper/DeferredResultWrapper.java]: How about just  `return Integer.compare(this.hashCode(), deferredResultWrapper.hashCode())`?

File: apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/wrapper/DeferredResultWrapper.java (hunk 1 above)
Comment (Accelerater) [reply apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/wrapper/DeferredResultWrapper.java]: yes

File: apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/wrapper/DeferredResultWrapper.java [hunk 4]
```diff
@@ -64,4 +65,9 @@ public void setResult(List<ApolloConfigNotification> notifications) {
   public DeferredResult<ResponseEntity<List<ApolloConfigNotification>>> getResult() {
     return result;
   }
+
+  @Override
+  public int compareTo(@NonNull DeferredResultWrapper deferredResultWrapper) {
+    return Integer.compare(this.hashCode() - deferredResultWrapper.hashCode(), 0);
```
Comment (nobodyiam) [inline apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/wrapper/DeferredResultWrapper.java]: I think this is ok, but is there any difference between `return Integer.compare(this.hashCode(), deferredResultWrapper.hashCode())`?

**Pull request merged**